docs/compose-file.md

```diff
@@ -12,79 +12,29 @@ parent="smn_compose_ref"
 
 # Compose file reference
 
-The compose file is a [YAML](http://yaml.org/) file where all the top level
-keys are the name of a service, and the values are the service definition.
-The default path for a compose file is `./docker-compose.yml`.
+The Compose file is a [YAML](http://yaml.org/) file defining
+[services](#service-configuration-reference),
+[networks](#network-configuration-reference) and
+[volumes](#volume-configuration-reference).
+The default path for a Compose file is `./docker-compose.yml`.
 
-Each service defined in `docker-compose.yml` must specify exactly one of
-`image` or `build`. Other keys are optional, and are analogous to their
-`docker run` command-line counterparts.
+A service definition contains configuration which will be applied to each
+container started for that service, much like passing command-line parameters to
+`docker run`. Likewise, network and volume definitions are analogous to
+`docker network create` and `docker volume create`.
```

Comment:
We have had a couple reports from users about it being difficult to find the docs about variable interpolation.  It's in the menu on the right, but the menu is so long it's easy to miss (since it's at the bottom).

It would probably be good to mention variable substitution as part of the intro paragraph here, that it's supported in all the sections of the config, and link to the section at the bottom.
It doesn't need to happen as part of these changes though.